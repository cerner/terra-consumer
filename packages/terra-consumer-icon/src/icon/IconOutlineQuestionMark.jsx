/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <circle fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" cx="24" cy="24" r="22" ></circle><circle data-color="color-2" fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" cx="24" cy="36" r="1" ></circle><path data-color="color-2" fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M24 29v-5c3.218 0 6-2.782 6-6s-2.782-6-6-6c-2.389 0-4.685 1.785-5.584 3.843" ></path><circle data-color="color-2" data-stroke="none" fill="#444" cx="24" cy="36" r="1" ></circle>
    </IconBase>
  );
};

SvgIcon.displayName = "IconOutlineQuestionMark";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","height":"48","width":"48","xmlns":"http://www.w3.org/2000/svg"};

export default SvgIcon;
/* eslint-enable */
