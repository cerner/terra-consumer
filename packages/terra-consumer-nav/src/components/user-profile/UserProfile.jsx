import React from 'react';
import PropTypes from 'prop-types';
import Arrange from 'terra-arrange';
import classNames from 'classnames/bind';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';
import IconEllipses from 'terra-icon/lib/icon/IconEllipses';
import IconOutlineUserCircle from 'terra-consumer-icon/lib/icon/IconOutlineUserCircle';
import ProfileLinks from './ProfileLinks';
import SafeHtml from '../safe-html/SafeHtml';
import styles from './UserProfile.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * User name to be displayed in profile if user is signed in.
   */
  userName: PropTypes.string,
  /**
   * Avatar to be displayed in profile.
   */
  avatar: PropTypes.element,
  /**
   * A unique id set to the profile popup button that will be referred in profile popup.
   */
  id: PropTypes.string,
  /**
   * The path signout button would redirect to.
   */
  signoutUrl: PropTypes.string.isRequired,
  /**
   * The path to the login page.
   */
  signinUrl: PropTypes.string,
  /**
   * Determiniate of whether profile should render as signin link or profile button.
   */
  isSignIn: PropTypes.bool,
  /**
   * The content of the each profile items.
   */
  profileLinks: PropTypes.array,
  /**
   * A function used as a callback to render modal and popup content.
   */
  handleClick: PropTypes.func.isRequired,
  /**
   * Injected react-intl formatting api
   */
  intl: intlShape.isRequired,
};

const defaultProps = {
  avatar: <IconOutlineUserCircle />,
  profileLinks: [],
  isSignIn: false,
  id: 'terra-conumser-nav-profile-button',
};

const UserProfile = ({
  userName, avatar, id, signoutUrl, signinUrl, isSignIn, profileLinks, handleClick, intl, ...customProps
}) => {
  let profileContent;
  if (isSignIn) {
    profileContent = (
      <a className={cx('popup-button')} href={signinUrl}>
        <Arrange
          fitStart={<div className={cx('avatar')}>{avatar}</div>}
          fill={<FormattedMessage id="Terra.Consumer.UserProfile.signin" />}
          align="center"
        />
      </a>
    );
  } else {
    const content = (
      <div>
        <ProfileLinks linkItems={profileLinks} />
        <a className={cx('link', 'signout-border')} href={signoutUrl}>
          <FormattedMessage id="Terra.Consumer.UserProfile.signout" />
        </a>
      </div>
    );

    const title = intl.formatMessage({ id: 'Terra.Consumer.UserProfile.Modal.title' });

    profileContent = (
      <button className={cx('popup-button')} onClick={() => handleClick({ title, content })}>
        <Arrange
          fitStart={<div className={cx('avatar')}>{avatar}</div>}
          fill={<SafeHtml text={userName} />}
          fitEnd={<IconEllipses className={cx('icon')} id={id} />}
          alignFitStart="top"
          alignFill="center"
          alignFitEnd="top"
        />
      </button>
    );
  }

  return (
    <div {...customProps} className={cx('profile')}>
      {profileContent}
    </div>
  );
};

UserProfile.propTypes = propTypes;
UserProfile.defaultProps = defaultProps;

export default injectIntl(UserProfile);
