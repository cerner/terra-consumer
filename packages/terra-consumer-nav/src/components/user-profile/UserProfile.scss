@import '../../variables';

:local {
  .profile {
    background: var(--terra-consumer-profile-link-background);
    background-attachment: fixed;
    background-size: cover;
    bottom: 0;
    position: fixed;
    width: var(--terra-consumer-profile-link-width, 320px);

    @media screen and (max-width: $terra-tiny-breakpoint) {
      width: 100%;
    }
  }

  .link {
    background: #fff;
    border: 0;
    border-radius: 0;
    color: #383f42;
    cursor: pointer;
    display: block;
    font-size: 14px;
    line-height: 18px;
    padding: 20px;
    text-align: left;
    text-decoration: none;
    width: 100%;

    &:hover,
    &:focus {
      background: #cfd0d1;
    }
  }

  .signout-border {
    @media screen and (min-width: $terra-medium-breakpoint) {
      border-top: 1px solid #dedfe0;
    }
  }

  .profile-item-border {
    @media screen and (max-width: $terra-medium-breakpoint) {
      border-bottom: 1px solid #dedfe0;
    }
  }

  .toggler-wrapper {
    border-bottom: 1px solid #dedfe0;

    @media screen and (min-width: $terra-medium-breakpoint) {
      border-bottom: 0;
      border-top: 1px solid #dedfe0;
    }

    &:active {
      color: inherit;
    }
  }

  .toggler {
    padding: 0;

    .link {
      padding: 15px 30px;

      &:not(:last-child) {
        border-bottom: 0;
      }
    }
  }

  /* stylelint-disable selector-max-compound-selectors */
  .open {
    .toggler-wrapper,
    .toggler .link {
      background: #f3f3f3;

      &:hover {
        background: #cfd0d1;
      }
    }
  }

  .popup-button {
    background: var(--terra-consumer-popup-button-background, rgba(255, 255, 255, 0.25));
    border: 0;
    border-radius: 0;
    color: var(--terra-consumer-profile-link-color, #3d3d3d);
    cursor: pointer;
    font-size: 16px;
    margin-top: 3px;
    padding: 20px 25px;
    text-align: left;
    width: inherit;

    &:hover,
    &:focus {
      background: rgba(255, 255, 255, 0.4);
      color: var(--terra-consumer-profile-link-color, #3d3d3d);
      text-decoration: underline;
    }
  }

  .avatar {
    padding-right: 10px;
  }

  .link-text-style {
    padding: 0;
    text-align: left;
    text-decoration: none;
  }
}
