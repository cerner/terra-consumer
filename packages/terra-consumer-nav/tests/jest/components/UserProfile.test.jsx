import React from 'react';
import IconPerson from 'terra-icon/lib/icon/IconPerson';
import IconOutlineQuestionMark from 'terra-consumer-icon/lib/icon/IconOutlineQuestionMark';
import UserProfile from '../../../src/components/user-profile/ProfileLinks';

const profileLinks = [
  {
    text: 'Account',
    icon: (<IconOutlineQuestionMark height={16} width={16} />),
    url: 'http://localhost:8080/',
  },
  {
    text: 'Notifications',
    icon: (<IconOutlineQuestionMark height={16} width={16} />),
    url: 'http://localhost:8080/',
  },
];

describe('UserProfile', () => {
  // Snapshot Tests
  it('should render a user profile with avatar,name and popup button', () => {
    const wrapper = shallow(<UserProfile
      profileLinks={profileLinks}
      name="Frank Lampard"
      avatar={<IconPerson />}
      signoutUrl="http://localhost:8080/"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
