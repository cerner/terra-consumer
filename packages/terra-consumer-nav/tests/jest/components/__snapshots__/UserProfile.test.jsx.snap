// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should render a user profile with avatar,name and popup button 1`] = `
<div
  avatar={
    <IconPerson
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  name="Frank Lampard"
  profileLinks={
    Array [
      Object {
        "icon": <IconOutlineQuestionMark
          height={16}
          viewBox="0 0 48 48"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
    />,
        "text": "Account",
        "url": "http://localhost:8080/",
      },
      Object {
        "icon": <IconOutlineQuestionMark
          height={16}
          viewBox="0 0 48 48"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
    />,
        "text": "Notifications",
        "url": "http://localhost:8080/",
      },
    ]
  }
  signoutUrl="http://localhost:8080/"
/>
`;
